src/Microsoft.AspNetCore.Sockets.Common/Internal/Formatters/ServerSentEventsMessageFormatter.cs

```diff
@@ -27,70 +30,53 @@ public static bool TryFormatMessage(Message message, Span<byte> buffer, out int
                 throw new InvalidOperationException("Cannot format message where endOfMessage is false using this format");
             }
 
-            // Need at least: Length of 'data: ', one character type, one \r\n, and the trailing \r\n
-            if (buffer.Length < DataPrefix.Length + 1 + Newline.Length + Newline.Length)
+            if (!TryGetTypeIndicator(message.Type, out var typeIndicator))
             {
-                bytesWritten = 0;
                 return false;
```

Comment:
Yep, I need to clean up some of the exceptions. Returning false is purely for "insufficient buffer space" errors. Format/parsing errors are exceptions. I'll review those.